### DISCUSSION TITLE
Feature Request: DNS rewrites based on IP

### DISCUSSION THREAD

**Discussion by webwude** [AdguardTeam/AdGuardHome#4998] [Ideas]
Hi.

I am using Adguard Home for quite some time now. I managed with reverse proxy to enable an access outside my LAN with HTTPS and certificates and even DNS over TLS with Android phones works.

The problem that I am facing: 
I use rewrites to let mydomain point to my server in my local network:
*.mydomain.com points to 10.0.0.250
which is working fine at home.

As soon as I am using DNS over TLS on my phone outside my LAN (e.g. with mobile data) the DNS resolver still tries to reroute it to 10.0.0.250 - which is not accessible outside my LAN.
Is there a way to disable rewrite if the IP is outside my LAN? This would be very handy.

Best regards
w.

**Comment by ainar-g** [comment]
Theoretically, something like this should work:

```adblock
||mydomain.com^$client=~10.0.0.0/8,dnsrewrite=NOERROR;A;12.34.56.78
||mydomain.com^$client=10.0.0.0/8,dnsrewrite=NOERROR;A;10.0.0.250
```

https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#client

**Reply by webwude** [reply]
Thx. This worked - almost to simple :)

**Reply by andekande** [reply]
please mark as answered
